example/extensions/lib_custom_op/gemm_lib.cc

```diff
@@ -140,6 +140,22 @@ MXReturnValue inferType(std::map<std::string, std::string> attrs,
   return MX_SUCCESS;
 }
 
+MXReturnValue inferSType(std::map<std::string, std::string> attrs,
+                        std::vector<int> &instypes,
+                        std::vector<int> &outstypes) {
+  // validate inputs
+  for (unsigned i = 0; i < instypes.size(); i++) {
+    if (instypes[i] != kDefaultStorage) {
+      std::cout << "Expected input " << i << " to have dense storage type" << std::endl;
+      return MX_FAIL;
+    }
+  }
+  for (unsigned i = 0; i < instypes.size(); i++) {
+    outstypes[i] = instypes[0];
```

Comment:
Is this going to work? you're looping over instypes.size and using that to index into outstypes. For Gemm i think we have 2 inputs and 1 output

Reply:
Fixed it.